lib/client-retry.js

```diff
@@ -0,0 +1,144 @@
+                    console.log(colors.red(`encountered a 429: ${e.message}.`));
+                    const waitTime = this.retryWaitTime(e);
+                    console.log(colors.red(`Waiting for ${waitTime}ms and trying again`));
+                    await sleep(waitTime);
+                } else {
+                    non429Response = true;
+                    error = e;
+                }
+            }
+        }
+        if (!non429Response) {
+            throw new Error("Running into 429s. Will not continue testing. Try re-running in a few minutes.");
+
+        }
+        // pass through error
+        if (error) {
+            throw error;
+        }
+        return response;
+    }
+
+    // retry on 429s
+    async unregister() {
+        let response, error;
+        let non429Response = false;
+        let retryCount = 0;
+        while (!non429Response && retryCount <= this.max429RetryCount) {
+            retryCount++;
+            try {
+                response = await super.unregister();
+                non429Response = true;
+            } catch (e) {
+                if (e.message.includes("429")) {
+                    console.log(colors.red(`encountered a 429: ${e.message}.`));
+                    const waitTime = this.retryWaitTime(e);
+                    console.log(colors.red(`Waiting for ${waitTime}ms and trying again`));
+                    await sleep(waitTime);
+                } else {
+                    non429Response = true;
+                    error = e;
+                }
+            }
+        }
+
+        if (!non429Response) {
+            throw new Error("Running into 429s. Will not continue testing. Try re-running in a few minutes.");
+
+        }
+        // pass through error
+        if (error) {
+            throw error;
+        }
+        return response;
+    }
+    async process(source, renditions, userData) {
+        let response, error;
+        let non429Response = false;
+        let retryCount = 0;
+        while (!non429Response && retryCount <= this.max429RetryCount) {
+            retryCount++;
+            try {
+                response = await super.process(source, renditions, userData);
+                console.log('Successful call to /process', retryCount);
+                non429Response = true;
+            } catch (e) {
+                if (e.message.includes("429")) {
+                    console.log(colors.red(`encountered a 429: ${e.message}.`));
+                    const waitTime = this.retryWaitTime(e);
+                    console.log(colors.red(`Waiting for ${waitTime}ms and trying again`));
+                    await sleep(waitTime);
+                } else {
+                    non429Response = true;
+                    error = e;
+                }
+            }
+        }
+
+        if (!non429Response) {
+            throw new Error("Running into 429s. Will not continue testing. Try re-running in a few minutes.");
```

Comment:
In case the library is not used in tests:
```suggestion
            throw new Error("Running into 429s. Will not continue retrying. Try again in a few minutes.");
```